[31650] Avoid builtin in user limit

oliverguenther opened the pull request:
https://community.openproject.com/wp/31650

oliverguenther committed:
[31650] Avoid builtin in user limit

https://community.openproject.com/wp/31650

machisuji reviewed (approved):
:+1: This should fix the issue, although I would appreciate it if you could add a regression spec for this scenario @oliverguenther . <3

machisuji committed:
added regression spec for active users and fixed anonymous user's status

machisuji commented:
@oliverguenther FYI I added the spec myself and fixed another issue which surfaced while running the spec. Once this is green we can merge it and release 10.2.1.

machisuji committed:
made deleted user locked; exclude builtin users from active_or_registered

Because active_or_registered are users which are meant to be worked with
in the system. We may want to rename this to something else.

machisuji committed:
anonymous is now active so we have to change this

machisuji committed:
skip builtin users in all counts (since anon and system are now active too, and deleted is blocked)

machisuji committed:
only 2 out of 3 builtins are actvie (deleted is blocked)

machisuji committed:
I'm sick of looking up magic numbers

machisuji committed:
fixed typo

oliverguenther committed:
Rewrite cuke
